From 766f21c91fbcc6ad14608409240552c24300e071 Mon Sep 17 00:00:00 2001
From: Sergei Shtylyov <sergei.shtylyov@cogentembedded.com>
Date: Fri, 19 Jul 2013 14:55:52 +0900
Subject: [PATCH 641/675] ARM: shmobile: r8a7778: fix SCR clock

According to R-Car M1A User's Manual rev. 1.00 en, SGX control register (SCR)
clock is controlled by bit 20 of the MSTPCR3 register, not bit 19.

Signed-off-by: Sergei Shtylyov <sergei.shtylyov@cogentembedded.com>
Signed-off-by: Yusuke Goda <yusuke.goda.sx@renesas.com>
---
 arch/arm/mach-shmobile/clock-r8a7778.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/arch/arm/mach-shmobile/clock-r8a7778.c b/arch/arm/mach-shmobile/clock-r8a7778.c
index 122a55c..5d3a70a 100644
--- a/arch/arm/mach-shmobile/clock-r8a7778.c
+++ b/arch/arm/mach-shmobile/clock-r8a7778.c
@@ -120,8 +120,8 @@ enum {
 	MSTP410,
 	MSTP408,
 	MSTP331,
-	MSTP323, MSTP322, MSTP321,
-	MSTP319, MSTP311, MSTP310, MSTP309, MSTP308, MSTP307,
+	MSTP323, MSTP322, MSTP321, MSTP320,
+	MSTP311, MSTP310, MSTP309, MSTP308, MSTP307,
 	MSTP114, MSTP110, MSTP109, MSTP107, MSTP105, MSTP103, MSTP100,
 	MSTP030,
 	MSTP029, MSTP028, MSTP027, MSTP026, MSTP025, MSTP024, MSTP023, MSTP022, MSTP021,
@@ -139,7 +139,7 @@ static struct clk mstp_clks[MSTP_NR] = {
 	[MSTP323] = SH_CLK_MSTP32(&p_clk, MSTPCR3, 23, 0), /* SDHI0 */
 	[MSTP322] = SH_CLK_MSTP32(&p_clk, MSTPCR3, 22, 0), /* SDHI1 */
 	[MSTP321] = SH_CLK_MSTP32(&p_clk, MSTPCR3, 21, 0), /* SDHI2 */
-	[MSTP319] = SH_CLK_MSTP32(&p_clk, MSTPCR3, 19, 0), /* SCR */
+	[MSTP320] = SH_CLK_MSTP32(&p_clk, MSTPCR3, 20, 0), /* SCR */
 	[MSTP311] = SH_CLK_MSTP32(&p_clk, MSTPCR3, 11, 0), /* SSI4 */
 	[MSTP310] = SH_CLK_MSTP32(&p_clk, MSTPCR3, 10, 0), /* SSI5 */
 	[MSTP309] = SH_CLK_MSTP32(&p_clk, MSTPCR3,  9, 0), /* SSI6 */
@@ -192,7 +192,7 @@ static struct clk_lookup lookups[] = {
 	CLKDEV_DEV_ID("sh_mobile_sdhi.0", &mstp_clks[MSTP323]), /* SDHI0 */
 	CLKDEV_DEV_ID("sh_mobile_sdhi.1", &mstp_clks[MSTP322]), /* SDHI1 */
 	CLKDEV_DEV_ID("sh_mobile_sdhi.2", &mstp_clks[MSTP321]), /* SDHI2 */
-	CLKDEV_CON_ID("scr",		&mstp_clks[MSTP319]), /* SCR */
+	CLKDEV_CON_ID("scr",		&mstp_clks[MSTP320]), /* SCR */
 	CLKDEV_DEV_ID("sh-eth",	&mstp_clks[MSTP114]), /* Ether */
 	CLKDEV_DEV_ID("r8a7778-vin.0", &mstp_clks[MSTP110]), /* VIN0 */
 	CLKDEV_DEV_ID("r8a7778-vin.1", &mstp_clks[MSTP109]), /* VIN1 */
-- 
1.8.2

